llvm/lib/RemoteCachingService/CMakeLists.txt

```diff
@@ -1,27 +1,32 @@
 if (LLVM_CAS_ENABLE_REMOTE_CACHE)
 add_subdirectory(RemoteCacheProto)
 add_subdirectory(RemoteCacheServer)
-add_subdirectory(Client)
-add_subdirectory(CAS)
 
 set(LLVM_LINK_COMPONENTS
   RemoteCacheKVProto
   RemoteCacheCASProto
-  RemoteCacheClient
-  GRPCRelayCAS
   )
-else ()
-add_subdirectory(NullService)
 
-set(LLVM_LINK_COMPONENTS
-  RemoteNullService
-  )
-endif()
+  include_directories($<TARGET_PROPERTY:LLVMRemoteCacheKVProto,BINARY_DIR>)
+  include_directories($<TARGET_PROPERTY:LLVMRemoteCacheCASProto,BINARY_DIR>)
 
+  add_llvm_component_library(LLVMRemoteCachingService
+  RemoteCachingService.cpp
+  Client/Client.cpp
```

Comment:
I suggest you only use one `add_llvm_component_library` and only set the sharing setting once.

You can declare different variable in the branch and set
```
add_llvm_component_library(LLVMRemoteCachingService
  $REMOTE_CACHE_SOURCE
  RemoteCachingService.cpp

  LINK_COMPONENTS
   Support
   CAS
)
```